var/spack/repos/builtin/packages/py-gevent/package.py

```diff
@@ -41,6 +41,8 @@ class PyGevent(PythonPackage):
 
     # https://github.com/gevent/gevent/issues/1599
     conflicts("^py-cython@3:", when="@:20.5.0")
+    # https://github.com/gevent/gevent/issues/2031
+    conflicts("^py-cython@3.0.10", when="@:23.9.1", msg="py-gevent fails to build when using cython@3.0.10")
```

Comment:
Upper limits on version ranges are inclusive in spack, so this would need something like
```suggestion
    conflicts("^py-cython@3.0.10", when="@:23.9.0", msg="py-gevent fails to build when using cython@3.0.10")
```
(That is if it fails to build with 23.9.0 but succeeds with 23.9.1.)